src/bluesearch/entrypoint/database/topic_extract.py

```diff
@@ -168,6 +173,33 @@ def run(
                     },
                 }
             )
+    elif source == "pubmed":
+        for path in inputs:
+            logger.info(f"Processing {path}")
+            articles = ElementTree.parse(str(input_path))
+            for i, article in enumerate(articles.iter("PubmedArticle")):
+                journal_topics, article_topics = get_topics_for_pubmed_article(article)
+                all_results.append(
+                    {
+                        "source": "pubmed",
+                        "path": str(path.resolve()),
+                        "topics": {
+                            "journal": {
+                                "MeSH": journal_topics,
+                            },
+                            "article": {
+                                "MeSH": article_topics,
+                            },
+                        },
+                        "metadata": {
+                            "created-date": datetime.datetime.now().strftime(
+                                "%Y-%m-%d %H:%M:%S"
+                            ),
+                            "bbs-version": bluesearch.version.__version__,
+                            "element_in_file": i,
+                        },
+                    }
+                )
```

Comment:
It seems we're starting to copy-paste this nested dictinoary with a fixed structure. I think it might be appropriate to factor it out into a separate data structure. We can probably try and adress this in a new issue.

The logic could be similar to the `Article` class we already have. So that we can write something like
```python
topic_info = TopicInfo()
topic_info.source = "pubmed"
topic_info.path = path.resolve()
topic_info.json() 
# gives the nested dictionary we're currently using, automatically fills the metadata part, etc.
```

Reply:
I totally agree! Sounds a good idea to open an issue/PR for this.

Reply:
Created #539

Reply:
Perfect! Thanks :)